arbnode/sequencer.go

```diff
@@ -631,21 +657,16 @@ func (s *Sequencer) precheckNonces(queueItems []txQueueItem) []txQueueItem {
 			// or higher than the highest tx nonce we've seen.
 			err := MakeNonceError(sender, txNonce, stateNonce)
 			if errors.Is(err, core.ErrNonceTooHigh) {
+				var nonceError NonceError
+				if !errors.As(err, &nonceError) {
+					log.Warn("unreachable nonce error is not nonceError")
```

Comment:
Could we have MakeNonceErrror return a NonceError instead of a generic error? I think that'd remove the need for this downcasting.

Reply:
the problem is with "nil" error.. tried something, LMK what you think